tests/functional/specs/test_multipackage1.spec

```diff
@@ -0,0 +1,62 @@
+# -*- mode: python -*-
+#-----------------------------------------------------------------------------
+# Copyright (c) 2013-2017, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License with exception
+# for distributing bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+
+# MULTIPROCESS FEATURE: file A (onefile pack) depends on file B (onefile pack).
+
+
+__testname__ = '../scripts/test_multipackage1'
+__testdep__ = '../scripts/multipackage1_B'
+
+a = Analysis([__testname__ + '.py'],
+             pathex=['.'])
+b = Analysis([__testdep__ + '.py'],
+             pathex=['.'])
+
+pyz = PYZ(a.pure, a.zipped_data,
+          b.pure, b.zipped_data,
+          append=False)
+
+exe = EXE(pyz,
+          a.scripts,
+          a.binaries,
+          a.zipfiles,
+          a.datas,
+          a.dependencies,
+          name=os.path.basename(__testname__),
+          debug=True,
+          strip=False,
+          upx=False,
+          console=1 )
+                    
+exeB = EXE(pyz,
+          b.scripts,
+          b.binaries,
+          b.zipfiles,
+          b.datas,
+          b.dependencies,
+          name=os.path.join('dist', __testdep__ + '.exe'),
+          debug=True,
+          strip=False,
+          upx=False,
+          console=1 )
+
+coll = COLLECT(
+        exe,
+        a.binaries,
```

Comment:
`COLLECT()` is not normally used for onefile exes. Specifically, passing `a.binaries` to `EXE()` above will include them into the exe as a onefile pack, and then passing `a.binaries` again to `COLLECT()` will include them a second time in the output folder as if this were a onedir pack. 

It will be unclear whether the build result is a onefile or onedir exe. It might even be unclear whether the multipackage tests are verifying that a dependency was found in a onefile exe or just found in the plain folder resulting from the COLLECT.

Try generating a pair of new specfiles using pyi-makespec, first with `--onefile` and then with `--onedir` to see the difference.